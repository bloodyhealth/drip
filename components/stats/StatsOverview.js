import React from 'react'
import { StyleSheet, View } from 'react-native'
import PropTypes from 'prop-types'

import AppText from '../common/app-text'

import { Sizes, Spacing, Typography } from '../../styles'

const StatsOverview = ({ data }) => {
  return data.map((rowContent, i) => <Row key={i} rowContent={rowContent} />)
}

StatsOverview.propTypes = {
  data: PropTypes.array.isRequired,
}

const Row = ({ rowContent }) => {
  const isStandardDeviation = rowContent[1].includes('deviation')

  return (
    <View style={styles.row}>
      <Cell content={rowContent[0]} isLeft />
      <Cell content={rowContent[1]} hasAsterisk={isStandardDeviation} />
    </View>
  )
}

Row.propTypes = {
  rowContent: PropTypes.array.isRequired,
}

const Cell = ({ content, isLeft, hasAsterisk }) => {
  const styleContainer = isLeft ? styles.cellLeft : styles.cellRight
  const styleText = isLeft ? styles.accentPurpleBig : styles.accentOrange
  const numberOfLines = isLeft ? 1 : 2
  const ellipsizeMode = isLeft ? 'clip' : 'tail'

  return (
    <View style={styleContainer}>
      <AppText
        numberOfLines={numberOfLines}
        ellipsizeMode={ellipsizeMode}
        style={styleText}
      >
        {content}
        {hasAsterisk && <AppText style={styles.accentOrange}>*</AppText>}
      </AppText>
    </View>
  )
}

Cell.propTypes = {
  content: PropTypes.node.isRequired,
  isLeft: PropTypes.bool,
  hasAsterisk: PropTypes.bool,
}

const styles = StyleSheet.create({
  accentOrange: {
    ...Typography.accentOrange,
    fontSize: Sizes.small,
    margin: Sizes.tiny,
  },
  accentPurpleBig: {
    ...Typography.accentPurpleBig,
    marginRight: Spacing.tiny,
  },
  cellLeft: {
    alignItems: 'flex-end',
    flex: 3,
    justifyContent: 'center',
  },
  cellRight: { flex: 5 },
  row: { flexDirection: 'row' },
})

export default StatsOverview
