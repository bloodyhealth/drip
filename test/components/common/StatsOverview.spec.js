import React from 'react'
import { render } from '@testing-library/react-native'

import StatsOverview from '../../../components/common/StatsOverview'

describe('StatsOverview screen', () => {
  test('when provided correct, renders it', async () => {
    const data = [
      [21, 'shortest'],
      [21, 'longest'],
      [0, 'standard deviation'],
      [2, 'completed cycles'],
    ]
    const { toJSON } = render(<StatsOverview data={data} />)

    expect(toJSON()).toMatchSnapshot()
  })

  test('when provided empty data, renders nothing (does not break)', async () => {
    const data = []
    const { toJSON } = render(<StatsOverview data={data} />)

    expect(toJSON()).toMatchSnapshot()
  })
})
