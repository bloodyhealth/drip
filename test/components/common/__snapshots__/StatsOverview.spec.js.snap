// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsOverview screen when provided correct, renders it 1`] = `
Array [
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 3,
          "justifyContent": "center",
        }
      }
    >
      <Text
        ellipsizeMode="clip"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#555",
              "fontFamily": "Jost-Book",
              "fontSize": 34.285714285714285,
            },
            Object {
              "color": "#3A2671",
              "fontFamily": "Jost-Bold",
              "fontSize": 64.28571428571428,
              "marginRight": 8.571428571428571,
              "textAlignVertical": "center",
              "textTransform": "uppercase",
            },
          ]
        }
      >
        21
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 5,
          "flexDirection": "row",
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Array [
            Object {
              "color": "#555",
              "fontFamily": "Jost-Book",
              "fontSize": 34.285714285714285,
            },
            Object {
              "color": "#F38337",
              "fontFamily": "Jost-Bold",
              "fontSize": 27.857142857142858,
              "margin": 15,
              "textAlignVertical": "center",
              "textTransform": "uppercase",
            },
          ]
        }
      >
        shortest
      </Text>
    </View>
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 3,
          "justifyContent": "center",
        }
      }
    >
      <Text
        ellipsizeMode="clip"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#555",
              "fontFamily": "Jost-Book",
              "fontSize": 34.285714285714285,
            },
            Object {
              "color": "#3A2671",
              "fontFamily": "Jost-Bold",
              "fontSize": 64.28571428571428,
              "marginRight": 8.571428571428571,
              "textAlignVertical": "center",
              "textTransform": "uppercase",
            },
          ]
        }
      >
        21
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 5,
          "flexDirection": "row",
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Array [
            Object {
              "color": "#555",
              "fontFamily": "Jost-Book",
              "fontSize": 34.285714285714285,
            },
            Object {
              "color": "#F38337",
              "fontFamily": "Jost-Bold",
              "fontSize": 27.857142857142858,
              "margin": 15,
              "textAlignVertical": "center",
              "textTransform": "uppercase",
            },
          ]
        }
      >
        longest
      </Text>
    </View>
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 3,
          "justifyContent": "center",
        }
      }
    >
      <Text
        ellipsizeMode="clip"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#555",
              "fontFamily": "Jost-Book",
              "fontSize": 34.285714285714285,
            },
            Object {
              "color": "#3A2671",
              "fontFamily": "Jost-Bold",
              "fontSize": 64.28571428571428,
              "marginRight": 8.571428571428571,
              "textAlignVertical": "center",
              "textTransform": "uppercase",
            },
          ]
        }
      >
        0
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 5,
          "flexDirection": "row",
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Array [
            Object {
              "color": "#555",
              "fontFamily": "Jost-Book",
              "fontSize": 34.285714285714285,
            },
            Object {
              "color": "#F38337",
              "fontFamily": "Jost-Bold",
              "fontSize": 27.857142857142858,
              "margin": 15,
              "textAlignVertical": "center",
              "textTransform": "uppercase",
            },
          ]
        }
      >
        standard deviation
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#555",
              "fontFamily": "Jost-Book",
              "fontSize": 34.285714285714285,
            },
            Object {
              "color": "#3A2671",
              "fontFamily": "Jost-Bold",
              "fontSize": 64.28571428571428,
              "marginRight": 8.571428571428571,
              "textAlignVertical": "center",
              "textTransform": "uppercase",
            },
          ]
        }
      >
        *
      </Text>
    </View>
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 3,
          "justifyContent": "center",
        }
      }
    >
      <Text
        ellipsizeMode="clip"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#555",
              "fontFamily": "Jost-Book",
              "fontSize": 34.285714285714285,
            },
            Object {
              "color": "#3A2671",
              "fontFamily": "Jost-Bold",
              "fontSize": 64.28571428571428,
              "marginRight": 8.571428571428571,
              "textAlignVertical": "center",
              "textTransform": "uppercase",
            },
          ]
        }
      >
        2
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 5,
          "flexDirection": "row",
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Array [
            Object {
              "color": "#555",
              "fontFamily": "Jost-Book",
              "fontSize": 34.285714285714285,
            },
            Object {
              "color": "#F38337",
              "fontFamily": "Jost-Bold",
              "fontSize": 27.857142857142858,
              "margin": 15,
              "textAlignVertical": "center",
              "textTransform": "uppercase",
            },
          ]
        }
      >
        completed cycles
      </Text>
    </View>
  </View>,
]
`;

exports[`StatsOverview screen when provided empty data, renders nothing (does not break) 1`] = `null`;
